addons/web/static/src/views/fields/properties/properties_field.js

```diff
@@ -594,6 +594,11 @@ export class PropertiesField extends Component {
      * @param {object} propertyDefinition
      */
     async onPropertyDefinitionChange(propertyDefinition) {
+        const propertyIndex = this._getPropertyIndex(propertyDefinition.name);
+        if (propertyIndex < 0) {
+            return;
```

Comment:
I've got a counterproposition for this fix:
<https://github.com/odoo-dev/odoo/commit/ec7b4b75182b40b67dfc1c4e2db1b654f75af01b>

because I think the real issue is in fact caused by `pick`, that is why the `propertyIndex` can sometimes not be found

The issue with your fix is that it just prevents the crash, it does not fix the underlying issue that the property_definition component state is incorrect because it is not aware of changes done after the `pick`

Reply:
For posterity, here is the link of the fix PR: https://github.com/odoo/odoo/pull/206871